js/components/channel_messages.jsx

```diff
@@ -36,6 +36,7 @@ var ChannelMessages = React.createClass({
   },
 
   componentWillUnmount: function() {
+    MessageStore.removeListener('change', this._messageStoreChange);
```

Comment:
I had to remove these listeners to prevent a React error when trying to unmount the component (while changing routes / switching component views). We do need to do some work to fix the 'two children with the same key warnings' (related to: https://github.com/rorysaur/quack/pull/53)